### PR TITLE
Usageevolution for Slingshot

### PR DESCRIPTION
**Pull request by stiesssh**
## Current

No UsageEvolution in Slingshot

## New

Slingshot can interprete UsageEvolution. 
This comprises:
* Option to (optionally) sprecifiy UsageEvolution model in the launch config. 
* Behaviour to trigger the updates based on events.
* Evolver-Classes (based on the SimuLizar Evolvers) that updated the evolvable RandomVariables in then UsageModel.

* **Beware** This PR also provides a minor change to make simulation of _open_ workloads reflect to the evolved values. However, there are no changes to make the simulation of _closed_ workloads refelct the evolved values, as i do not know how to implement that properly.

## Misc
This PR is mostly about updating the underlying PCM Model, *not* about making sure, that the simulation considers the updates.
Only exception: minor change regarding openWorkloads.

### CODE REVIEW

File: org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+					openwl.setSpecification(newRateStr);
+				}
+			} else if (wl instanceof ClosedWorkload) {
+				final int newRateInt = (int) Math.round(newRate);
+				final int oldRate = ((ClosedWorkload) wl).getPopulation();
+				if (newRateInt == oldRate) {
+					LOGGER.debug("Closed workload population is still: " + newRateInt);
+				} else {
+					LOGGER.debug("Changing closed workload population from: " + oldRate + " to " + newRateInt);
+					((ClosedWorkload) wl).setPopulation(newRateInt);
```
Comment (klinakuf) [inline org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java:148]: This does not affect the simulation at the moment. The users are spawned on Simulation Start.

**Review by klinakuf — CHANGES_REQUESTED** [review]

File: org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java (hunk 1 above)
Comment (stiesssh) [reply org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java:149]: True. I cannt do anything about that though.

File: org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/UsageEvolutionBehavior.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import de.uka.ipd.sdq.simucomframework.SimuComConfig;
+
+/**
+ * The System simulation behavior is a extension that simulates the system
+ * model. It listens to events requesting to interpret the repository and
+ * sometimes will result in a SEFF Interpretation request if there is a RDSeff.
+ *
+ * @author Julijan Katic
+ */
```
[the remark this answers is not in the document]
Comment (stiesssh) [reply org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/UsageEvolutionBehavior.java:40]: fixed it.

File: org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java (hunk 1 above)
Comment (klinakuf) [reply org.palladiosimulator.analyzer.slingshot.behavior.usageevolution/src/org/palladiosimulator/analyzer/slingshot/behavior/usageevolution/evolver/AbstractUsageEvolver.java:149]: Maybe throwing, in addition, a ModelChanged event for both, while here the implementation assumes that somebody will pick it up correctly. It's like in SPD; when the interpreter changes the model by transformations, it throws events. Then it should be up to the other extensions to say look, if this part changes, then indeed I need to do something or state hey, I am stateless. I don't care about this change.

**Review by klinakuf — APPROVED** [review]
Implementation works and the code structure looks fine. A seperate issue has to be created for the interpretretation of changes in the closed workload. 

I have tested the implementation with the BIS model where I change both parameters for the evolution step width as well as the level of workload.

**Pull request merged**